.github/workflows/examples-infineon.yaml

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build example - Infineon P6
+
+on:
+    push:
+    pull_request:
+    workflow_dispatch:
+
+concurrency:
+    group: ${{ github.ref }}-${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.event.number) || (github.event_name == 'workflow_dispatch' && github.run_number) || github.sha }}
+    cancel-in-progress: true    
+
+jobs:
+    infineon:
+        name: Infineon examples building
+        timeout-minutes: 30
+        
+        env:
+            BUILD_TYPE: P6
+            
+        runs-on: ubuntu-latest
+        if: github.actor != 'restyled-io[bot]'
+
+        container:
+            image: connectedhomeip/chip-build-infineon:latest
+            volumes:
+                - "/tmp/output_binaries:/tmp/output_binaries"
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
+
+            - name: Build lock-app example
+              timeout-minutes: 10
+              run: scripts/examples/gn_p6_example.sh
```

Comment:
could you add p6 support into `./scripts/build/build_examples.py` ? That way this line becomes generic (rather than calling a p6-specific shell script) and we will get vscode integration for test purposes somewhat for free (after #9052 lands).

Reply:
Added infineon support in build_examples.py script